lib/common/tray_handler.dart

```diff
@@ -0,0 +1,87 @@
+import 'dart:ui';
+
+import 'package:lantern/common/common.dart';
+import 'package:lantern/common/common_desktop.dart';
+import 'package:tray_manager/tray_manager.dart';
+import 'package:window_manager/window_manager.dart';
+
+String systemTrayIcon(bool connected) {
+  if (connected) {
+    return Platform.isWindows
+        ? 'assets/images/lantern_connected_32.ico'
+        : 'assets/images/lantern_connected_32.png';
+  }
+  return Platform.isWindows
+      ? 'assets/images/lantern_disconnected_32.ico'
+      : 'assets/images/lantern_disconnected_32.png';
+}
+
+
+class TrayHandler with TrayListener {
+  TrayHandler() {
+    if (isDesktop()) {
+      setupTray(false);
+    }
+  }
+
+  Future<void> setupTray(bool isConnected) async {
+    await trayManager.setIcon(systemTrayIcon(isConnected));    
+    Menu menu = Menu(
+      items: [
+        MenuItem(
+          key: 'status',
+          disabled: true,
+          label: isConnected ? 'status_on'.i18n : 'status_off'.i18n,
+        ),
+        MenuItem(
+          key: 'status',
+          label: isConnected ? 'disconnect'.i18n : 'connect'.i18n,
+        ),
+        MenuItem.separator(),
+        MenuItem(
+          key: 'show_window',
+          label: 'show'.i18n,
+        ),
+        MenuItem.separator(),
+        MenuItem(
+          key: 'exit',
+          label: 'exit'.i18n,
+        ),
+      ],
+    );
+    await trayManager.setContextMenu(menu);
+  }
+
+  @override
+  void onTrayMenuItemClick(MenuItem menuItem) {
+    switch (menuItem.key) {
+      case 'show':
+        windowManager.show();
```

Comment:
@atavism I think we should use `windowManager.focus();` instead of  `windowManager.show();`

Reply:
Not sure I understand the difference (show is supposed to show and give focus on the window) but I can change it

Reply:
So if you have multiple windows open on Mac then if you click to show it does not show up foreground, I think we talked about this before, when I working on some desktop fixes, that's why I remember,